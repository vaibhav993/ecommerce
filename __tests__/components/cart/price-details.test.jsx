import { render, screen } from "@testing-library/react"

import PriceDetails from '@/components/cart/price-details'

describe('PriceDetails', () => {

  const createTestProps = (props={}) => ({
    cartItems: [
      {
        id: 1,
        quantity: 2,
        itemData: {
          "id": 1,
          "name": "Green Polo",
          "price": 250,
          "currency": "INR",
          "quantity": 1
        }
      },
      {
        id: 1,
        quantity: 1,
        itemData: {
          "id": 1,
          "name": "Green Polo",
          "price": 250,
          "currency": "INR",
          "quantity": 1
        }
      },
    ],
    ...props
  })

  let props = createTestProps()
  
  it('renders PriceDetails', () => {
    const { container } = render(<PriceDetails {...props}/>)
    expect(container).toMatchSnapshot()
  })

  it('- should render correct data', () => {
    render(<PriceDetails {...props}/>)
    const totalMRP = screen.getByTestId('mrp-price')
    const totalAmount = screen.getByTestId('total-amount')
    expect(totalMRP).toHaveTextContent('₹750.00')
    expect(totalAmount).toHaveTextContent('₹750.00')
  })
})