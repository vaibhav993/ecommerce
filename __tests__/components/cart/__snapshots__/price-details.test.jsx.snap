// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceDetails renders PriceDetails 1`] = `
<div>
  <div
    class="price-details"
  >
    <h4
      data-testid="price-details-title"
    >
      Price Details (2 items)
    </h4>
    <div
      class="base-row"
    >
      <span>
        Total MRP
      </span>
      <span
        data-testid="mrp-price"
      >
        ₹750.00
      </span>
    </div>
    <div
      class="base-row"
    >
      <span>
        Discount
      </span>
      <span
        data-testid="discount"
      >
        -₹0.00
      </span>
    </div>
    <div
      class="base-row"
    >
      <span>
        Delivery Charges
      </span>
      <span
        data-testid="delivery-charges"
      >
        FREE
      </span>
    </div>
    <div
      class="base-total"
    >
      <span>
        Total Amount
      </span>
      <span
        data-testid="total-amount"
      >
        ₹750.00
      </span>
    </div>
  </div>
</div>
`;
